src/derive/curve.rs

```diff
@@ -145,15 +145,238 @@ macro_rules! new_curve_impl {
     (($($privacy:tt)*),
     $name:ident,
     $name_affine:ident,
-    $name_compressed:ident,
-    $compressed_size:expr,
+    $flags_extra_byte:expr,
     $base:ident,
     $scalar:ident,
     $generator:expr,
     $constant_b:expr,
     $curve_id:literal,
     ) => {
 
+        macro_rules! impl_compressed {
+            () => {
+                paste::paste! {
+
+                #[allow(non_upper_case_globals)]
+                const [< $name _COMPRESSED_SIZE >]: usize = if $flags_extra_byte {$base::size() + 1} else {$base::size()};
+                // const [< $name _COMPRESSED_SIZE >]: usize = 32;
+                #[derive(Copy, Clone)]
+                pub struct [<$name Compressed >]([u8; [< $name _COMPRESSED_SIZE >]]);
+
+                // Compressed
+                impl std::fmt::Debug for [< $name Compressed >] {
+                    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+                        self.0[..].fmt(f)
+                    }
+                }
+
+                impl Default for [< $name Compressed >] {
+                    fn default() -> Self {
+                        [< $name Compressed >]([0; [< $name _COMPRESSED_SIZE >]])
+                    }
+                }
+
+                impl AsRef<[u8]> for [< $name Compressed >] {
+                    fn as_ref(&self) -> &[u8] {
+                        &self.0
+                    }
+                }
+
+                impl AsMut<[u8]> for [< $name Compressed >] {
+                    fn as_mut(&mut self) -> &mut [u8] {
+                        &mut self.0
+                    }
+                }
+
+                impl group::GroupEncoding for $name_affine {
+                    type Repr = [< $name Compressed >];
+
+                    fn from_bytes(bytes: &Self::Repr) -> CtOption<Self> {
+                        Self::from_bytes_unchecked(bytes).and_then(|p| CtOption::new(p, p.is_on_curve()))
+                    }
+
+                    fn from_bytes_unchecked(bytes: &Self::Repr) -> CtOption<Self> {
+                        let bytes = &bytes.0;
+                        let mut tmp = *bytes;
+                        let is_inf = Choice::from(tmp[[< $name _COMPRESSED_SIZE >] - 1] >> 7);
+                        let ysign = Choice::from(tmp[[< $name _COMPRESSED_SIZE >] - 1] >> 6);
```

Comment:
So now this only supports curve with field size less than 254 bit (since we are using the second most significant bit as y sign).

Reply:
Yeah make sense.